client/configuration_variable_test.go

```diff
@@ -0,0 +1,179 @@
+package client_test
+
+import (
+	. "github.com/env0/terraform-provider-env0/client"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Configuration Variable", func() {
+	mockConfigurationVariable := ConfigurationVariable{
+		Id:             "config-var-id-789",
+		Name:           "configName",
+		Value:          "configValue",
+		OrganizationId: organizationId,
+		IsSensitive:    true,
+		Scope:          "PROJECT",
+		Type:           0,
+		ScopeId:        "project-123",
+		UserId:         "user|123",
+	}
+
+	Describe("ConfigurationVariableCreate", func() {
+		var createdConfigurationVariable ConfigurationVariable
+
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId)
+
+			expectedCreateRequest := []map[string]interface{}{{
+				"name":           mockConfigurationVariable.Name,
+				"isSensitive":    mockConfigurationVariable.IsSensitive,
+				"value":          mockConfigurationVariable.Value,
+				"organizationId": organizationId,
+				"scopeId":        mockConfigurationVariable.ScopeId,
+				"scope":          Scope(mockConfigurationVariable.Scope),
+				"type":           ConfigurationVariableType(mockConfigurationVariable.Type),
+			}}
+
+			httpCall = mockHttpClient.EXPECT().
+				Post("configuration", expectedCreateRequest, gomock.Any()).
+				Do(func(path string, request interface{}, response *[]ConfigurationVariable) {
+					*response = []ConfigurationVariable{mockConfigurationVariable}
+				})
+
+			createdConfigurationVariable, _ = apiClient.ConfigurationVariableCreate(
+				mockConfigurationVariable.Name,
+				mockConfigurationVariable.Value,
+				mockConfigurationVariable.IsSensitive,
+				Scope(mockConfigurationVariable.Scope),
+				mockConfigurationVariable.ScopeId,
+				ConfigurationVariableType(mockConfigurationVariable.Type),
+				nil,
+			)
+		})
+
+		It("Should get organization id", func() {
+			organizationIdCall.Times(1)
+		})
+
+		It("Should send POST request with params", func() {
```

Comment:
it seems like that this It only check that httpCall has been called one time, maybe we should add assert for the expected params (`toHaveBeenCalledWith`). 
if we stay with httpCall.Times(1) maybe it will be more clear to remove "with params" from the It description

Reply:
oh I missed the `mockHttpClient.EXPECT()` in line 40.
you can ignore the comment above

Reply:
yeah I don't like it either but that's how gomock works ...
@yaronya any way we could move the assertion from the `Before` to the `It`?

Reply:
That's how `gomock` works and how mocking in golang works in general, we can't compare it to JS 🤷 

The mock is created per function arguments so this has to be done **prior** to function call. For the sake of nicer BDD, I've used the `Times()` function to assert it on the `It` clause so in case it fails, the test output log would include it.